hash-tables/ex1/ex1.py

```diff
@@ -13,7 +13,24 @@ def get_indices_of_item_weights(weights, length, limit):
     YOUR CODE HERE
     """
 
-    return None
+    if length == 1:
+        result = None
+
+    for i in range(length):
+        hash_table_insert(ht, weights[i], i)
+
+    for i in range(length):
+        package_1 = weights[i]
+        package_2 = limit - package_1
+        search = hash_table_retrieve(ht, package_2)
+
+        if search is not None:
+            if search >= i:
+                result = (search, i)
+            elif search < i:
+                result = (i, search)
+
+    return result
 
 
```

Comment:
Hey, Bryce, you mentioned you had a test `test_ex1_2` not passing. It looks like the code is returning `(1,1)` with that test expecting `(1,0)`. I think what's happening is that both weights are equal to the _same_ number (4) so the culprit may be your logic in the `if/elif` block that causing a problem for that test. Another way to fix this could be to `break` out of the loop after you've assigned the matching `tuple` to `result`, and sorting by `desc` order 👍